cpp/src/arrow/memory_pool.cc

```diff
@@ -201,4 +201,45 @@ int64_t LoggingMemoryPool::max_memory() const {
   std::cout << "max_memory: " << mem << std::endl;
   return mem;
 }
+
+ProxyMemoryPool::ProxyMemoryPool(MemoryPool* pool) : pool_(pool), proxy_bytes_allocated_(0) {}
+
+Status ProxyMemoryPool::Allocate(int64_t size, uint8_t** out) {
+  proxy_bytes_allocated_ += size;
+  Status s = pool_->Allocate(size, out);
+  std::cout << "Allocate: size = " << size << std::endl;
```

Comment:
Please remove the calls to `std::cout`. If the user wants this, he should explicitly chose the `LoggingMemoryPool`.